pkg/reconciler/revision/revision.go

```diff
@@ -106,17 +106,21 @@ func (c *Reconciler) reconcileDigest(ctx context.Context, rev *v1.Revision) erro
 	}
 
 	digests := make(chan digestData, len(rev.Spec.Containers))
+	d := digestData{}
```

Comment:
it seems what you really want here is:
1. an error variable (you don't really need the whole digest object)
2. sync.Once to enure the error is just set once. Right now the last error wins, but any of them should do.